### PR TITLE
Upgrade to Hibernate ORM 5.4.22.Final and matching Hibernate Reactive 1.0.0.Alpha10

### LINKED ISSUE #12004: Hibernate ORM build error during enhancement of entity
**Describe the bug**
I encountered a build failure of my project during the enhancement phase of an entity, which is caused by not being able to access an attribute from its parent class.

The hierarchie of the involved entities is as follows:

```java
 @MappedSuperclass
 public abstract class AutoId<T extends AutoId<T>> {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    public Integer getId() {
        return id;
    }

    public T setId(Integer id) {
        this.id = id;
        return (T) this;
    }

}
```

```java
 @MappedSuperclass
 public abstract class TaskComponent<T extends TaskComponent<T>>
        extends AutoId<T> {

    @Basic
    @Column(name = "name")
    private String name;

    @Basic
    @Column(name = "tags",
            columnDefinition = "TEXT")
    @Convert(converter = TagAttributeConverter.class)
    private List<Tag> tags;

    public String getName() {
        return name;
    }

    public T setName(String name) {
        this.name = name;
        return (T) this;
    }

    public List<Tag> getTags() {
        if (Objects.isNull(this.tags)) {
            this.tags = new ArrayList<>();
        }
        return tags;
    }

    public T setTags(List<Tag> tags) {
        this.tags = tags;
        return (T) this;
    }

}
```
The `tags` attribute is causing the issue.

```java
 @Entity
 @Table(name = "measurement_descriptor",
       schema = "test")
public class MeasurementDescriptor
        extends TaskComponent<MeasurementDescriptor>
        implements Serializable {

    private static final long serialVersionUID = -6908645489234141976L;

    @ManyToOne
    @JoinColumn(name = "unit_id")
    private Unit unit;

    public Unit getUnit() {
        return unit;
    }

    public MeasurementDescriptor setUnit(Unit unit) {
        this.unit = unit;
        return this;
    }

}
```
```java
public class Tag {

    private String name;
    private String value;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getValue() {
        return value;
    }

    public void setValue(String value) {
        this.value = value;
    }

}
```
```java
@Converter
public class TagAttributeConverter
        implements AttributeConverter<List<Tag>, String> {

    @Override
    public String convertToDatabaseColumn(List<Tag> attribute) {
        return "empty";
    }

    @Override
    public List<Tag> convertToEntityAttribute(String dbData) {
        return Collections.emptyList();
    }

}
```
```java
@Entity
@Table(name = "unit",
       schema = "test")
public class Unit
        extends AutoId<Unit> {

    @Column(name = "name")
    private String name;

    @Column(name = "symbol")
    private String symbol;

    public String getName() {
        return name;
    }

    public Unit setName(String name) {
        this.name = name;
        return this;
    }

    public String getSymbol() {
        return symbol;
    }

    public Unit setSymbol(String symbol) {
        this.symbol = symbol;
        return this;
    }

}
```

The maven build output is as follows
```
Failed to execute goal io.quarkus:quarkus-maven-plugin:1.7.2.Final:build (default) on project quarkus-test: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.deployment.steps.ClassTransformingBuildStep#handleClassTransformation threw an exception: java.lang.IllegalStateException: java.util.concurrent.ExecutionException: org.hibernate.bytecode.enhance.spi.EnhancementException: Failed to enhance class com.test.MeasurementDescriptor
	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:941)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
	at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
	at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2046)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1578)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
	at java.base/java.lang.Thread.run(Thread.java:834)
	at org.jboss.threads.JBossThread.run(JBossThread.java:479)
Caused by: java.util.concurrent.ExecutionException: org.hibernate.bytecode.enhance.spi.EnhancementException: Failed to enhance class com.test.MeasurementDescriptor
	at java.base/java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:191)
	at io.quarkus.deployment.steps.ClassTransformingBuildStep.handleClassTransformation(ClassTransformingBuildStep.java:122)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:932)
	... 7 more
Caused by: org.hibernate.bytecode.enhance.spi.EnhancementException: Failed to enhance class com.test.MeasurementDescriptor
	at org.hibernate.bytecode.enhance.internal.bytebuddy.EnhancerImpl.enhance(EnhancerImpl.java:138)
	at io.quarkus.hibernate.orm.deployment.HibernateEntityEnhancer$HibernateEnhancingClassVisitor.hibernateEnhancement(HibernateEntityEnhancer.java:81)
	at io.quarkus.hibernate.orm.deployment.HibernateEntityEnhancer$HibernateEnhancingClassVisitor.visitEnd(HibernateEntityEnhancer.java:74)
	at org.objectweb.asm.ClassReader.accept(ClassReader.java:722)
	at org.objectweb.asm.ClassReader.accept(ClassReader.java:401)
	at io.quarkus.deployment.steps.ClassTransformingBuildStep$1.call(ClassTransformingBuildStep.java:102)
	at io.quarkus.deployment.steps.ClassTransformingBuildStep$1.call(ClassTransformingBuildStep.java:82)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.IllegalStateException: Cannot access private java.util.List com.test.TaskComponent.tags from class com.test.MeasurementDescriptor
	at net.bytebuddy.asm.Advice$OffsetMapping$ForField$Resolved.resolve(Advice.java:2355)
	at net.bytebuddy.asm.Advice$OffsetMapping$ForField.resolve(Advice.java:2130)
	at net.bytebuddy.asm.Advice$Dispatcher$Inlining$Resolved$ForMethodExit.doApply(Advice.java:8062)
	at net.bytebuddy.asm.Advice$Dispatcher$Inlining$Resolved$ForMethodExit.apply(Advice.java:8023)
	at net.bytebuddy.asm.Advice$Dispatcher$Inlining$Resolved$AdviceMethodInliner.visitMethod(Advice.java:7500)
	at net.bytebuddy.jar.asm.ClassReader.readMethod(ClassReader.java:1327)
	at net.bytebuddy.jar.asm.ClassReader.accept(ClassReader.java:718)
	at net.bytebuddy.jar.asm.ClassReader.accept(ClassReader.java:401)
	at net.bytebuddy.asm.Advice$Dispatcher$Inlining$Resolved$AdviceMethodInliner.apply(Advice.java:7494)
	at net.bytebuddy.asm.Advice$AdviceVisitor$WithExitAdvice.onUserEnd(Advice.java:9890)
	at net.bytebuddy.asm.Advice$AdviceVisitor.visitMaxs(Advice.java:9670)
	at net.bytebuddy.asm.Advice$Appender$EmulatingMethodVisitor.resolve(Advice.java:10241)
	at net.bytebuddy.asm.Advice$Appender.apply(Advice.java:10193)
	at net.bytebuddy.dynamic.scaffold.TypeWriter$MethodPool$Record$ForDefinedMethod$WithBody.applyCode(TypeWriter.java:708)
	at net.bytebuddy.dynamic.scaffold.TypeWriter$MethodPool$Record$ForDefinedMethod$WithBody.applyBody(TypeWriter.java:693)
	at net.bytebuddy.dynamic.scaffold.TypeWriter$MethodPool$Record$ForDefinedMethod.apply(TypeWriter.java:600)
	at net.bytebuddy.dynamic.scaffold.TypeWriter$Default$ForInlining$WithFullProcessing$RedefinitionClassVisitor.onVisitEnd(TypeWriter.java:5022)
	at net.bytebuddy.utility.visitor.MetadataAwareClassVisitor.visitEnd(MetadataAwareClassVisitor.java:323)
	at net.bytebuddy.jar.asm.ClassReader.accept(ClassReader.java:722)
	at net.bytebuddy.jar.asm.ClassReader.accept(ClassReader.java:401)
	at net.bytebuddy.dynamic.scaffold.TypeWriter$Default$ForInlining.create(TypeWriter.java:3827)
	at net.bytebuddy.dynamic.scaffold.TypeWriter$Default.make(TypeWriter.java:2166)
	at net.bytebuddy.dynamic.scaffold.inline.RedefinitionDynamicTypeBuilder.make(RedefinitionDynamicTypeBuilder.java:224)
	at net.bytebuddy.dynamic.DynamicType$Builder$AbstractBase.make(DynamicType.java:3588)
	at net.bytebuddy.dynamic.DynamicType$Builder$AbstractBase$Delegator.make(DynamicType.java:3833)
	at org.hibernate.bytecode.internal.bytebuddy.ByteBuddyState.make(ByteBuddyState.java:197)
	at org.hibernate.bytecode.internal.bytebuddy.ByteBuddyState.rewrite(ByteBuddyState.java:146)
	at org.hibernate.bytecode.enhance.internal.bytebuddy.EnhancerImpl.enhance(EnhancerImpl.java:132)
	... 10 more
```

**Expected behavior**
Successful maven build.

**Actual behavior**
Build fails. (A work around is to change the access modifier of the attribute `tags` from `private` to `protected` or `public`. Then the maven build completes successfully.)

**To Reproduce**
1.  It should be enough to create a simple quarkus project with those listed classes and try to build the project with maven

**Configuration**
```properties
# Add your application.properties here, if applicable.
quarkus.datasource.username=postgres
quarkus.datasource.password=postgres
quarkus.datasource.driver=org.postgresql.Driver
quarkus.datasource.url=jdbc:postgresql://localhost:5432/quarkus

quarkus.hibernate-orm.database.generation=drop-and-create
```

**Screenshots**
(If applicable, add screenshots to help explain your problem.)

**Environment (please complete the following information):**
 - OS: Microsoft Windows 10
 - Output of `java -version`: 
```
openjdk version "14.0.2" 2020-07-14
OpenJDK Runtime Environment AdoptOpenJDK (build 14.0.2+12)
Eclipse OpenJ9 VM AdoptOpenJDK (build openj9-0.21.0, JRE 14 Windows 10 amd64-64-Bit 20200715_147 (JIT enabled, AOT enabled)
OpenJ9   - 34cf4c075
OMR      - 113e54219
JCL      - 1d231bd6a2 based on jdk-14.0.2+12)
```
 - GraalVM version (if different from Java): 
`graalvm-ce-java11-windows-amd64-20.2.0`
 - Quarkus version or git rev: `1.7.2.Final`
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): 
`Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)`

**Additional context**
(Add any other context about the problem here.)
pom.xml

```maven
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
    <modelVersion>4.0.0</modelVersion>
    <groupId>com.test</groupId>
    <artifactId>quarkus-test</artifactId>
    <version>0.1</version>
    <packaging>jar</packaging>
    <properties>
        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
        <maven.compiler.source>11</maven.compiler.source>
        <maven.compiler.target>11</maven.compiler.target>
        <quarkus.version>1.7.2.Final</quarkus.version>
    </properties>
    
    <dependencyManagement>
        <dependencies>
            <dependency>
                <groupId>io.quarkus</groupId>
                <artifactId>quarkus-bom</artifactId>
                <version>${quarkus.version}</version>
                <type>pom</type>
                <scope>import</scope>
            </dependency>
        </dependencies>
    </dependencyManagement>
    
    <dependencies>        
        <!-- Hibernate ORM specific dependencies -->
        <dependency>
            <groupId>io.quarkus</groupId>
            <artifactId>quarkus-hibernate-orm</artifactId>
            <version>${quarkus.version}</version>
        </dependency>

        <!-- JDBC driver dependencies -->
        <dependency>
            <groupId>io.quarkus</groupId>
            <artifactId>quarkus-jdbc-postgresql</artifactId>
            <version>${quarkus.version}</version>
        </dependency>
    </dependencies>
    
    <build>
        <plugins>
            <plugin>
                <groupId>io.quarkus</groupId>
                <artifactId>quarkus-maven-plugin</artifactId>
                <version>${quarkus.version}</version>
                <executions>
                    <execution>
                        <goals>
                            <goal>build</goal>
                        </goals>
                    </execution>
                </executions>
            </plugin>
        </plugins>
    </build>
    
    <profiles>
        <profile>
            <id>native</id>
            <build>
                <plugins>
                    <plugin>
                        <groupId>io.quarkus</groupId>
                        <artifactId>quarkus-maven-plugin</artifactId>
                        <version>${quarkus.version}</version>
                        <executions>
                            <execution>
                                <goals>
                                    <goal>native-image</goal>                                    
                                </goals>
                                <configuration>
                                    <enableHttpUrlHandler>true</enableHttpUrlHandler>
                                    <additionalBuildArgs>--report-unsupported-elements-at-runtime</additionalBuildArgs>
                                </configuration>
                            </execution>
                        </executions>
                    </plugin>
                    <plugin>
                        <groupId>org.apache.maven.plugins</groupId>
                        <artifactId>maven-failsafe-plugin</artifactId>
                        <version>2.22.2</version>                            
                        <executions>
                            <execution>
                                <goals>
                                    <goal>integration-test</goal>
                                    <goal>verify</goal>
                                </goals>
                                <configuration>
                                    <systemProperties>
                                        <native.image.path>${project.build.directory}/${project.build.finalName}-runner</native.image.path>
                                    </systemProperties>
                                </configuration>
                            </execution>
                        </executions>
                    </plugin>
                </plugins>
            </build>
        </profile>
    </profiles>
</project>
```

### PR DESCRIPTION
**Pull request by Sanne**
Release notes - Hibernate ORM - Version 5.4.22

** Bug
    * [HHH-9422] - Metamodel Generator should close streams opened to persistence.xml and referenced mapping files
    * [HHH-13058] - Criteria API correlated subquery with outer join generates incorrect SQL
    * [HHH-13201] - FromElement orign check fails when fetching @ElementCollection and association
    * [HHH-14148] - Invalid SQL when null precedence, @OrderBy and entity graph are used
    * [HHH-14197] - Criteria API doubly-nested subquery generates invalid SQL - missing subquery root
    * [HHH-14199] - setDataBase gradle task failed
    * [HHH-14201] - Wrong order of joins in SQL
    * [HHH-14206] - Auto-apply converter not applied to primitives
    * [HHH-14212] - @EntityGraph annotation log warning message "Entity graph specified is not applicable to the entity ... . Ignored." and works as if  EntityGraph.EntityGraphType.LOAD would set.
    * [HHH-14213] - Could not parse literal [9223372036854775808] as integer
    * [HHH-14219] - Duplicated rows of sequence table if generator shared by multiple entities
    * [HHH-14220] - EnhancementException failure with MappedSuperclass with a un-mapped collection
    * [HHH-14223] - NullPointerException when implicit joining over @JoinFormula/@JoinColumnOrFormula
    * [HHH-14230] - ClassCastException raised by ForeignKeyKey.toString()
    * [HHH-14231] - java.lang.ClassCastException: [B cannot be cast to [Ljava.lang.Object; while using JPA 2.2 query.getResultStream().findFirst()

** New Feature
    * [HHH-14198] - Expose CompositeUserTypes through JPA Metamodel

** Task
    * [HHH-14203] - DB2 requires the restrict keyword to drop the schema
    * [HHH-14204] - Schema validator should be more forgiving of column type mismatches
    * [HHH-14217] - Use ANSI escape codes to highlight output of hibernate.show_sql

** Improvement
    * [HHH-14031] - Update h2 database from 1.4.196 to 1.4.197
    * [HHH-14205] - Replace StringBuffer usage with StringBuilder when possible
    * [HHH-14218] - Correct some typos in code comments
    * [HHH-14236] - Improve efficiency of ForeignKeyKey
    * [HHH-14238] - Option to always include collection fields in the default fetch group
    * [HHH-14242] - Micro optimisations of ActionQueue: redundant branch evaluation

Fixes #12004

### CODE REVIEW

**Comment by Sanne** [comment]
@gavinking / @cescoffier / @aguibert : if any of you have a moment for an easy review, I'd love to merge this quickly.

All CI tests are green, should be easy unless you spot something silly :)

To be followed by https://github.com/quarkusio/quarkus-quickstarts/pull/680

**Review by cescoffier — APPROVED** [review]
The mutiny usage looks good to me, but I can't say anything about the "persistence" part.

**Comment by Sanne** [comment]
thanks @cescoffier ! There's no other important changes than the Mutiny API usage, so I'll merge.

File: integration-tests/hibernate-reactive-postgresql/src/main/java/io/quarkus/it/hibernate/reactive/postgresql/HibernateReactiveTestEndpointAlternative.java [hunk 1]
```diff
@@ -45,32 +45,32 @@
     public Uni<GuineaPig> reactiveFindMutiny() {
         final GuineaPig expectedPig = new GuineaPig(5, "Aloi");
         return populateDB()
-                .onItem().transformToUni(junk -> mutinySession.find(GuineaPig.class, expectedPig.getId()));
+                .chain(() -> mutinySession.find(GuineaPig.class, expectedPig.getId()));
```
Comment (aguibert) [inline integration-tests/hibernate-reactive-postgresql/src/main/java/io/quarkus/it/hibernate/reactive/postgresql/HibernateReactiveTestEndpointAlternative.java:48]: nice!

**Pull request merged**